drivers/overlay/overlay.go

```diff
@@ -97,18 +100,41 @@ func (d *driver) configure() error {
 			d.store, err = datastore.NewDataStore(datastore.GlobalScope, cfg)
 			if err != nil {
 				err = fmt.Errorf("failed to initialize data store: %v", err)
-				return
 			}
 		}
-
-		d.vxlanIdm, err = idm.New(d.store, "vxlan-id", vxlanIDStart, vxlanIDEnd)
-		if err != nil {
-			err = fmt.Errorf("failed to initialize vxlan id manager: %v", err)
-			return
-		}
 	})
 
-	return err
+	if err != nil {
+		return err
+	}
+
+	if d.store == nil {
+		return types.NoServiceErrorf("datastore is not available")
+	}
+
+	if d.vxlanIdm == nil {
+		return d.initializeVxlanIdm()
+	}
+
```

Comment:
It's because I wanted to retain current lazy logic where idm is initialized at network creation time.
Initialization of the vxlan idm now has to happen outside of the `Once` block given the only time the Once block is executed may not have initialized idm (no datastore provided at boot).